contracts/crypto-verify/src/contract.rs

```diff
@@ -62,7 +73,41 @@ pub fn query_verify_cosmos(
     let hash = Sha256::digest(message);
 
     // Verification
-    let result = deps.api.secp256k1_verify(&*hash, signature, public_key);
+    let result = deps
+        .api
+        .secp256k1_verify(hash.as_ref(), signature, public_key);
+    match result {
+        Ok(verifies) => Ok(VerifyResponse { verifies }),
+        Err(err) => Err(err.into()),
+    }
+}
+
+pub fn query_verify_ethereum(
+    deps: Deps,
+    message: &[u8],
+    signature: &[u8],
+    signer_address: &str,
+) -> StdResult<VerifyResponse> {
+    // Hashing
+    let mut hasher = Keccak256::new();
+    hasher.update(format!("\x19Ethereum Signed Message:\n{}", message.len()));
+    hasher.update(message);
+    let hash = hasher.finalize();
+
+    // Decompose signature
+    let (v, rs) = match signature.split_last() {
+        Some(pair) => pair,
+        None => return Err(StdError::generic_err("Signature must not be empty")),
+    };
+    let recovery = v - 27;
+
+    // Verification
+    let calculated_pubkey = deps.api.secp256k1_recover_pubkey(&hash, rs, recovery)?;
```

Comment:
Because the encoding of the recovery param is highly application specific. This `v` being 27/28 instead of 0/1 is an Ethereum thing. [They also encode the chain ID into that byte](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md).

The API intends to provide the pure alorithm where the recovery param is one of 0, 1.

Reply:
Thanks for the explanation.